crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -1885,15 +1887,85 @@ async fn test_entry_point_string() {
         "entry_point_string",
         "utf8_arg",
         vec![],
+        vec![
+            TestCallArg::Pure(utf_str_bcs),
+            TestCallArg::Pure(bcs::to_bytes(&n).unwrap()),
+        ],
+    )
+    .await
+    .unwrap();
+    assert_eq!(effects.status(), &ExecutionStatus::Success);
+}
+
+#[tokio::test]
+#[cfg_attr(msim, ignore)]
+async fn test_nested_string() {
+    let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+    let gas = ObjectID::random();
+    let authority = init_state_with_ids(vec![(sender, gas)]).await;
+
+    let package = build_and_publish_test_package(
+        &authority,
+        &sender,
+        &sender_key,
+        &gas,
+        "entry_point_string",
+    )
+    .await;
+
+    // pass an option utf8 string
+    let utf8_str = Some("çå∞≠¢õß∂ƒ∫");
+    let utf_str_bcs = bcs::to_bytes(&utf8_str).unwrap();
+    let effects = call_move(
+        &authority,
+        &gas,
+        &sender,
+        &sender_key,
+        &package.0,
+        "entry_point_string",
+        "option_utf8_arg",
+        vec![],
         vec![TestCallArg::Pure(utf_str_bcs)],
     )
     .await
     .unwrap();
-    assert!(
-        matches!(effects.status(), ExecutionStatus::Success { .. }),
-        "{:?}",
-        effects.status()
-    );
+    assert_eq!(effects.status(), &ExecutionStatus::Success);
+
+    // vector option utf8 string
+    let utf8_str = vec![Some("çå∞≠¢õß∂ƒ∫")];
+    let utf_str_bcs = bcs::to_bytes(&utf8_str).unwrap();
+    let effects = call_move(
+        &authority,
+        &gas,
+        &sender,
+        &sender_key,
+        &package.0,
+        "entry_point_string",
+        "vec_option_utf8_arg",
+        vec![],
+        vec![TestCallArg::Pure(utf_str_bcs)],
+    )
+    .await
+    .unwrap();
+    assert_eq!(effects.status(), &ExecutionStatus::Success);
+
+    // vector option utf8 string
```

Comment:
You should add a test where layout is option but someone passes 2 items in a vec